[CI] SSLDriverTests testHandshakeFailureBecauseProtocolMismatch failing

**Build Scans:**
- [elasticsearch-periodic #4653 / corretto8_java-matrix](https://gradle-enterprise.elastic.co/s/we7fo2raj4itg)
- [elasticsearch-periodic #4653 / corretto11_java-matrix](https://gradle-enterprise.elastic.co/s/ssr3rrwr63p3i)
- [elasticsearch-periodic #4648 / corretto11_java-matrix](https://gradle-enterprise.elastic.co/s/e6fstgta6t7am)
- [elasticsearch-periodic #4648 / corretto8_java-matrix](https://gradle-enterprise.elastic.co/s/ls4do5rmjrcie)
- [elasticsearch-periodic #4642 / corretto11_java-matrix](https://gradle-enterprise.elastic.co/s/ilnwanrfw4zcm)
- [elasticsearch-periodic #4642 / corretto8_java-matrix](https://gradle-enterprise.elastic.co/s/2x772cisv3hqo)
- [elasticsearch-periodic #4637 / corretto8_java-matrix](https://gradle-enterprise.elastic.co/s/nrwl5jevkp6dy)
- [elasticsearch-periodic #4637 / corretto11_java-matrix](https://gradle-enterprise.elastic.co/s/2p3ptpatmaweq)
- [elasticsearch-periodic #4632 / corretto11_java-matrix](https://gradle-enterprise.elastic.co/s/vmkjllr62xydk)
- [elasticsearch-periodic #4632 / corretto8_java-matrix](https://gradle-enterprise.elastic.co/s/jilccxxfeazoq)

**Reproduction Line:**
```
./gradlew ':x-pack:plugin:security:test' --tests "org.elasticsearch.xpack.security.transport.nio.SSLDriverTests.testHandshakeFailureBecauseProtocolMismatch" -Dtests.seed=E882D1342B4415C -Dtests.locale=es-SV -Dtests.timezone=US/Pacific -Druntime.java=8
```

**Applicable branches:**
7.17

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.security.transport.nio.SSLDriverTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(testHandshakeFailureBecauseProtocolMismatch),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: (is "The client supported protocol versions [TLSv1.1] are not accepted by server preferences [TLS1.2]" or is "Client requested protocol TLSv1.1 not enabled or not supported" or is "No appropriate protocol (protocol is disabled or cipher suites are inappropriate)")
     but: was "The client supported protocol versions [TLSv1.1] are not accepted by server preferences [TLS12]"
```

**Issue Reasons:**
- [7.17] 2 consecutive failures in test testHandshakeFailureBecauseProtocolMismatch
- [7.17] 5 consecutive failures in step corretto8_java-matrix
- [7.17] 20 consecutive failures in step corretto11_java-matrix
- [7.17] 25 failures in test testHandshakeFailureBecauseProtocolMismatch (3.2% fail rate in 774 executions)
- [7.17] 5 failures in step corretto8_java-matrix (23.8% fail rate in 21 executions)
- [7.17] 20 failures in step corretto11_java-matrix (100.0% fail rate in 20 executions)
- [7.17] 20 failures in pipeline elasticsearch-periodic (95.2% fail rate in 21 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/transport/nio/SSLDriverTests.java
+++ b/x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/transport/nio/SSLDriverTests.java
@@ -172,7 +172,7 @@ public class SSLDriverTests extends ESTestCase {
             serverProtocols = new String[] { "TLSv1.2" };
             clientProtocols = new String[] { "TLSv1.1" };
             expectedMessageMatcher = anyOf(
-                is("The client supported protocol versions [TLSv1.1] are not accepted by server preferences " + "[TLS12]"),
+                is("The client supported protocol versions [TLSv1.1] are not accepted by server preferences " + "[TLS1.2]"),
                 is("org.bouncycastle.tls.TlsFatalAlert: protocol_version(70)"),
                 is("Client requested protocol TLSv1.1 " + "not enabled or not supported")
             );
@@ -185,7 +185,7 @@ public class SSLDriverTests extends ESTestCase {
             serverProtocols = new String[] { "TLSv1.2" };
             clientProtocols = new String[] { "TLSv1.1" };
             expectedMessageMatcher = anyOf(
-                is("The client supported protocol versions [TLSv1.1] are not accepted by server preferences [TLS12]"),
+                is("The client supported protocol versions [TLSv1.1] are not accepted by server preferences [TLS1.2]"),
                 is("Client requested protocol TLSv1.1 not enabled or not supported"),
                 is("No appropriate protocol (protocol is disabled or cipher suites are inappropriate)")
             );

